src/equations/shallow_water_exner_1d.jl

```diff
@@ -0,0 +1,618 @@
+  associated energy\
+  [DOI: 10.1051/m2an/2016018](https://doi.org/10.1051/m2an/2016018)
+"""
+struct ShieldsStressModel{RealT} <: SedimentModel{RealT}
+    m_1::RealT
+    m_2::RealT
+    m_3::RealT
+    k_1::RealT
+    k_2::RealT
+    k_3::RealT
+    theta_c::RealT    # critical shields stress
+    d_s::RealT        # grain diameter
+end
+
+@doc raw"""
+    GrassModel(; A_g, m_g=3)
+
+Creates a Grass model to compute the sediment discharge `q_s` as
+```math
+q_s = A_g v^{m_g}
+```
+with the coefficients `A_g` and `m_g`.
+
+An overview of different formulations to compute the sediment discharge can be found in:
+- M.J. Castro Díaz, E.D. Fernández-Nieto, A.M. Ferreiro (2008)\
+  Sediment transport models in Shallow Water equations and numerical approach by high order 
+  finite volume methods\
+  [DOI:10.1016/j.compfluid.2007.07.017](https://doi.org/10.1016/j.compfluid.2007.07.017)
+"""
+struct GrassModel{RealT} <: SedimentModel{RealT}
+    A_g::RealT
+    m_g::RealT
+end
+
+function GrassModel(; A_g, m_g = 3.0)
+    return GrassModel(A_g, m_g)
+end
+
+@doc raw"""
+    MeyerPeterMueller(; theta_c, d_s)
+
+Creates a Meyer-Peter-Mueller model to compute the sediment discharge 
+`q_s` with the critical Shields stress `theta_c` and the grain diameter `d_s`.
+
+An overview of different formulations to compute the sediment discharge can be found in:
+- M.J. Castro Díaz, E.D. Fernández-Nieto, A.M. Ferreiro (2008)\
+  Sediment transport models in Shallow Water equations and numerical approach by high order 
+  finite volume methods\
+  [DOI:10.1016/j.compfluid.2007.07.017](https://doi.org/10.1016/j.compfluid.2007.07.017)
+"""
+function MeyerPeterMueller(; theta_c, d_s)
+    return ShieldsStressModel(0.0, 1.5, 0.0, 8.0, 1.0, 0.0, theta_c, d_s)
+end
+
+@doc raw"""
+    ShallowWaterExnerEquations1D(;gravity_constant, H0 = 0.0,
+                                 friction = ManningFriction(n = 0.0),
+                                 sediment_model,
+                                 porosity,
+                                 rho_f, rho_s)
+Entropy conservative formulation of the Shallow water-Exner equations in one space dimension. 
+The equations are given by
+```math
+\begin{cases}
+\partial_t h + \partial_x hv = 0, \\
+\partial_t hv + \partial_x (hv^2) + gh\partial_x (h + h_b) + g\frac{1}{r}h_s\partial_x (rh + h_b) + \frac{\tau}{\rho_f} = 0,\\
+\partial_t h_b + \partial_x q_s = 0,
+\end{cases}
+```
+The unknown quantities are the water and sediment height ``h``, ``h_b`` and the velocity ``v``.
+The sediment discharge ``q_s(h, hv)`` is determined by the `sediment_model` and is used to determine 
+the active sediment height ``h_s = q_s / v``. 
+Furthermore ``\tau`` denotes the shear stress at the water-sediment interface and is determined by 
+the `friction` model.  
+The gravitational constant is denoted by ``g``, and ``\rho_f`` and ``\rho_s`` are the fluid and sediment
+densities, respectively. The density ratio is given by ``r = \rho_f / \rho_s``.
```

Comment:
Yes, otherwise the sediment would be lighter than the fluid which is unphysical. I added an additional sentence about this in https://github.com/trixi-framework/TrixiShallowWater.jl/pull/51/commits/f73f6e980707b325869773513dfc0f07da633710